pydantic/_internal/_std_types_schema.py

```diff
@@ -109,67 +116,82 @@ def decimal_schema(_schema_generator: GenerateSchema, _decimal_type: type[Decima
         js_metadata=JsonSchemaMetadata(
             source_class=UUID, type='string', format='uuid', modify_js_function=lambda schema: schema.pop('anyOf', None)
         )
     )
-    # TODO, is this actually faster than `function_after(union(is_instance, is_str, is_bytes))`?
-    return core_schema.union_schema(
-        core_schema.is_instance_schema(uuid_type),
-        core_schema.function_after_schema(
-            core_schema.union_schema(
-                core_schema.str_schema(),
-                core_schema.bytes_schema(),
+    return core_schema.lax_or_strict_schema(
+        # TODO, is this actually faster than `function_after(union(is_instance, is_str, is_bytes))`?
+        lax_schema=core_schema.union_schema(
+            core_schema.is_instance_schema(uuid_type),
+            core_schema.function_after_schema(
+                core_schema.union_schema(
+                    core_schema.str_schema(),
+                    core_schema.bytes_schema(),
+                ),
+                _validators.uuid_validator,
+                metadata=metadata,
             ),
-            _validators.uuid_validator,
-            metadata=metadata,
+            custom_error_type='uuid_type',
+            custom_error_message='Input should be a valid UUID, string, or bytes',
+            strict=True,
         ),
-        custom_error_type='uuid_type',
-        custom_error_message='Input should be a valid UUID, string, or bytes',
-        strict=True,
+        strict_schema=core_schema.is_instance_schema(uuid_type),
     )
 
 
 @schema_function(PurePath)
-def path_schema(_schema_generator: GenerateSchema, path_type: type[PurePath]) -> core_schema.UnionSchema:
+def path_schema(_schema_generator: GenerateSchema, path_type: type[PurePath]) -> core_schema.LaxOrStrictSchema:
     metadata = build_metadata_dict(js_metadata=JsonSchemaMetadata(source_class=PurePath, type='string', format='path'))
-    # TODO, is this actually faster than `function_after(...)` as above?
-    return core_schema.union_schema(
-        core_schema.is_instance_schema(path_type),
-        core_schema.function_after_schema(
-            core_schema.str_schema(),
-            _validators.path_validator,
-            metadata=metadata,
+    return core_schema.lax_or_strict_schema(
+        # TODO, is this actually faster than `function_after(...)` as above?
+        lax_schema=core_schema.union_schema(
+            core_schema.is_instance_schema(path_type),
+            core_schema.function_after_schema(
+                core_schema.str_schema(),
+                _validators.path_validator,
+                metadata=metadata,
+            ),
+            custom_error_type='path_type',
+            custom_error_message='Input is not a valid path',
+            strict=True,
         ),
-        custom_error_type='path_type',
-        custom_error_message='Input is not a valid path',
-        strict=True,
+        strict_schema=core_schema.is_instance_schema(path_type),
```

Comment:
again we can add `json_types={'str'}`, but we will need to add an after validator to validate the input is valid if it came from JSON.